exercises/practice/parallel-letter-frequency/.meta/proof.ci.js

```diff
@@ -1,18 +1,48 @@
-export const parallelLetterFrequency = (texts) => {
-  let result = {};
-  let formatedTexts = texts
-    .map((x) => x.toLowerCase().match(/\p{Letter}+/gu) ?? [])
-    .flat();
-  Promise.all(formatedTexts.map((t) => processSingleText(t, result)));
-  return result;
-};
+const {
+  Worker,
+  isMainThread,
+  workerData,
+  parentPort,
+} = require('node:worker_threads');
 
-const processSingleText = (text, result) => {
-  return new Promise((resolve) => {
-    let res = [...text].reduce((acc, cur) => {
+if (isMainThread) {
+  module.exports.parallelLetterFrequency = async function (texts) {
+    let formatedTexts = texts
+      .map(
+        (x) =>
+          x
+            .toLowerCase()
+            .match(/\p{Letter}+/gu)
```

Comment:
Wouldn't it be better for the formatting to be also delegated to each worker? It's work (takes resources), and is done independently to each text.

Reply:
Sure, but as thus is just the CI proof, I don't really think it's necessary.

Also, in practice, I expect workers to be slower than just regular concurrency, since they're not as optimized, so it's debatable how much more performance you'd get from them.

Reply:
Oh yeah, I don't think we are doing this to be performant but just to show people that workers work for JS on node!

Reply:
OK then :)